submissions/bilalix/solution.py

```diff
@@ -0,0 +1,70 @@
+import csv
+from collections import defaultdict
+
+"""
+This script is a desperate attempt to preserve Python's dignity
+even though we know it won't beat C++ nor Rust ;)
+'Python hada jhdi 3lik! ¯\_(ツ)_/¯'
+"""
+
+def read_and_process_data(input_file_path):
+    """
+    Reads the CSV file and processes the data to determine 
+    the cheapest prices per product in each city
+    """
+    # This dictionary is where we will be storing the cheapest prices and city totals
+    city_totals = defaultdict(float)
+    city_products = defaultdict(lambda: defaultdict(lambda: float('inf')))
+    
+    # Bismillah..
+    with open(input_file_path, mode='r', newline='') as file:
```

Comment:
this will load the whole file in memory and trigger an OOM

Reply:
it shouldn't because it does not actually read the file content into memory; it simply opens the file in read mode, and the file is read line by line using `for .. in ..`